### PR TITLE
user.Current and sudo check

### PR DESCRIPTION
**Pull request by azilber**
Sudo check and partial switch to user.Current

### CODE REVIEW

**Comment by azilber** [comment]
This is a cleanup of https://github.com/fishworks/gofish/pull/186

**Review by bacongobbler — CHANGES_REQUESTED** [review]
can you write some tests to cover these changes? Perhaps some refactoring is in order.

File: cmd/gofish/ensure_unix.go [hunk 1]
```diff
@@ -6,31 +7,35 @@ import (
...
 		return err
 	}
-	// strip the newline character from the end
-	curUser := strings.TrimSuffix(string(userOutput), "\n")
 
-	groupCmd := exec.Command("id", "-gn", curUser)
-	groupOutput, err := groupCmd.Output()
-	if err != nil {
-		return err
+	if i, _ := strconv.Atoi(curUser.Uid); i == 0 {
```
[the remark this answers is not in the document]
Comment (azilber) [reply cmd/gofish/ensure_unix.go]: fixed

File: cmd/gofish/ensure_unix.go [hunk 2]
```diff
@@ -6,31 +7,35 @@ import (
...
+		isRoot = true
 	}
-	// strip the newline character from the end
-	curGroup := strings.TrimSuffix(string(groupOutput), "\n")
+
+	curGroup, _ := user.LookupGroupId(curUser.Gid)
```
[the remark this answers is not in the document]
Comment (azilber) [reply cmd/gofish/ensure_unix.go]: done

**Comment by azilber** [comment]
> can you write some tests to cover these changes? Perhaps some refactoring is in order.

Unfortunately I don't have the bandwidth right now to setup any testing.  This was a simple quality of life patch.

**Pull request closed without merging**